server.go

```diff
@@ -17,6 +18,27 @@ const (
 	fallbackCount     = 100
 )
 
+// getFilter returns a validated filter if present in the url query, nil otherwise.
+func getFilter(r *http.Request, s schema.Schema, extensions ...schema.Schema) (filter.Expression, error) {
+	rawFilter := strings.TrimSpace(r.URL.Query().Get("filter"))
+	if rawFilter == "" {
+		return nil, nil // No filter present.
+	}
+
+	decodedFilter, err := url.QueryUnescape(rawFilter)
+	if err != nil {
+		return nil, err
+	}
+	validator, err := f.NewValidator(decodedFilter, s, extensions...)
+	if err != nil {
+		return nil, err
+	}
+	if validator, err = f.NewValidator(decodedFilter, s, extensions...); err != nil {
+		return nil, err
+	}
```

Comment:
Shouldn't we call `Validate` here instead?

Reply:
Thanks for catching that!